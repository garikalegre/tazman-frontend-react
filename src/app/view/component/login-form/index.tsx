import React, { FC, FormEvent } from 'react';
import { Form as AntForm, Icon, Input, Button } from 'antd';
import { WrappedFormUtils } from 'antd/lib/form/Form';

type FormData = {
    email: string;
    password: string;
};

type Props = {
    form: WrappedFormUtils;
    onSubmit: (formData: FormData) => void;
};

const Form: FC<Props> = ({ form, onSubmit }) => {
    const { getFieldDecorator } = form;

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        form.validateFields((err, { email, password }) => {
            if (!err) {
                if (password.length < 8)
                    form.setFields({
                        password: {
                            value: password,
                            errors: [
                                new Error(
                                    'Password must be at least 8 characters'
                                )
                            ]
                        }
                    });
                else {
                    onSubmit({ email, password });
                }
            }
        });
    };

    return (
        <AntForm onSubmit={handleSubmit} className="login-form">
            <AntForm.Item>
                {getFieldDecorator('email', {
                    rules: [
                        {
                            required: true,
                            message: 'Please input your Email!'
                        }
                    ]
                })(
                    <Input
                        prefix={
                            <Icon
                                type="mail"
                                style={{ color: 'rgba(0,0,0,.25)' }}
                            />
                        }
                        type="email"
                        placeholder="Email"
                    />
                )}
            </AntForm.Item>
            <AntForm.Item>
                {getFieldDecorator('password', {
                    rules: [
                        {
                            required: true,
                            message: 'Please input your Password!'
                        }
                    ]
                })(
                    <Input
                        prefix={
                            <Icon
                                type="lock"
                                style={{ color: 'rgba(0,0,0,.25)' }}
                            />
                        }
                        type="password"
                        placeholder="Password"
                    />
                )}
            </AntForm.Item>
            <AntForm.Item>
                <Button
                    type="primary"
                    htmlType="submit"
                    className="login-form-button"
                >
                    Log in
                </Button>
            </AntForm.Item>
        </AntForm>
    );
};

export const LoginForm = AntForm.create<any>()(Form);
