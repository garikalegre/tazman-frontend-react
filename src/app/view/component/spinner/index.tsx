import React from 'react';
import { Spin } from 'antd';

export const Spinner = () => (
    <div className="spinner">
        <Spin size="large" />
    </div>
);
