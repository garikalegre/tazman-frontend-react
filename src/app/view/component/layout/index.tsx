import React, { FC, useEffect } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { RootActions, RootState } from '../../../model/types';
import * as AuthActions from '../../../duck/auth/auth.action';
import { Spinner } from '../spinner';

type Props = {
    checkAuth: () => void;
    isLoading: boolean;
};
const LayoutComponent: FC<Props> = ({ children, checkAuth, isLoading }) => {
    useEffect(() => {
        checkAuth();
    }, [checkAuth]);
    return <main>{isLoading ? <Spinner /> : children}</main>;
};

const mapStateToProps = (state: RootState) => ({
    isLoading: state.auth.isLoading
});
const mapDispatchToProps = (dispatch: Dispatch<RootActions>) => ({
    checkAuth: () => dispatch(AuthActions.checkAuthRequest())
});
export const Layout = connect(
    mapStateToProps,
    mapDispatchToProps
)(LayoutComponent);
