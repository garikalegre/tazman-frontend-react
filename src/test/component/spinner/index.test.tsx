import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from '../../../app/view/component/spinner';

describe('Spinner component', () => {
    it('render correctly', () => {
        const wrapper = shallow(<Spinner />);
        expect(wrapper).toMatchSnapshot();
    });
});
