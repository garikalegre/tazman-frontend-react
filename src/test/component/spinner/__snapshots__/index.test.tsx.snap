// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component render correctly 1`] = `
<div
  className="spinner"
>
  <Spin
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</div>
`;
